packages/commonwealth/client/scripts/state/api/launchPad/buyToken.ts

```diff
@@ -28,9 +33,55 @@ const buyToken = async ({
   return await launchPad.buyToken(amountEth, walletAddress, `${ethChainId}`);
 };
 
+export const handleQuerySuccess = async (
+  _: unknown,
+  variables: Omit<BuyTokenProps, 'amountEth'>,
+  trpcUtils: ReturnType<typeof trpc.useUtils>,
+) => {
+  await queryClient.invalidateQueries({
+    queryKey: getUserEthBalanceQueryKey({
+      chainRpc: variables.chainRpc,
+      ethChainId: variables.ethChainId,
+      walletAddress: variables.walletAddress,
+    }),
+  });
+  await queryClient.invalidateQueries({
+    queryKey: getERC20BalanceQueryKey({
+      tokenAddress: variables.tokenAddress,
+      userAddress: variables.walletAddress,
+      nodeRpc: variables.chainRpc,
+    }),
+  });
+  await queryClient.invalidateQueries({
+    queryKey: getTokenEthExchangeRateQueryKey({
+      chainRpc: variables.chainRpc,
+      ethChainId: variables.ethChainId,
+      mode: 'buy',
+      tokenAddress: variables.tokenAddress,
+      tokenAmount: 1 * 1e18, // amount per unit
+    }),
+  });
+  await queryClient.invalidateQueries({
+    queryKey: getTokenEthExchangeRateQueryKey({
+      chainRpc: variables.chainRpc,
+      ethChainId: variables.ethChainId,
+      mode: 'sell',
+      tokenAddress: variables.tokenAddress,
+      tokenAmount: 1 * 1e18, // amount per unit
+    }),
+  });
+  await trpcUtils.token.getTokens.invalidate();
+  await trpcUtils.token.getTokens.refetch();
+  await trpcUtils.token.getToken.invalidate();
```

Comment:
do we need this in this order? like invalidating it, refetching and invalidating again?

Reply:
this (trpc invalidation) doesn't actually work, so need to figure it out.

Reply:
can you try `queryClient.invalidateQueries` instead of trpcUtils?

Reply:
i that that but didnt work, made a different attempt with `queryClient.invalidateQueries` + `predicate` and it seems to call the APIs when invalidated but those new responses don't return fresh data. Maybe the issue is the data coming from API needs a few milliseconds to update (not entirely sure). Will test this in QA env to make sure.

For now will close this pr as it includes some other changes.